### PR TITLE
Qingwei/debug perf

### PR DESCRIPTION
**Pull request by qingweibinary**
@borisyankov 

Profiling on macbook I am using does not show significant difference, although number of render is reduce a whole lot. Time taken and profile chart both are close

each render spent more than 80% of CPU time in binary, maybe that's a better place to optimize

Note: This is for discussion, do not merge unless you're sure ;)

### CODE REVIEW

**Comment by borisyankov** [comment]
> each render spent more than 80% of CPU time in binary

What does that mean? :)

**Comment by qingweibinary** [comment]
@borisyankov typo

in `binary-charts`

**Comment by borisyankov** [comment]
Hmmm...
I do know why we don't see that much improvement ;(

File: src/trade/trade-chart/TradeViewChart.js [hunk 1]
```diff
@@ -67,6 +67,16 @@ export default class TradeViewChart extends PureComponent {
         }
     }
 
+    shouldComponentUpdate(nextProps) {
+        const hasNotEq = Object.keys(this.props).some(k => {
+            if (JSON.stringify(this.props[k]) !== JSON.stringify(nextProps[k])) {
```
Comment (borisyankov) [inline src/trade/trade-chart/TradeViewChart.js]: This is not fast.

**Pull request closed without merging**